src/main/java/net/snowflake/client/core/structs/SnowflakeObjectTypeFactories.java

```diff
@@ -0,0 +1,29 @@
+package net.snowflake.client.core.structs;
+
+import static net.snowflake.client.util.Predicates.notNull;
+
+import java.sql.SQLData;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public class SnowflakeObjectTypeFactories {
+  private static final Map<Class<?>, Supplier<SQLData>> factories = new HashMap<>();
+
+  public static synchronized void register(Class<?> type, Supplier<SQLData> factory) {
```

Comment:
do we need to have synchronized method here or can we use ConcurrentHashMap instead?